conn/physic/units.go

```diff
@@ -1001,14 +1050,18 @@ func noUnits(s string) error {
 	return &parseError{s: s, err: errors.New("no units provided, need")}
 }
 
+func incorrectUnit(inputString, want string) error {
+	return &parseError{err: errors.New("\"" + inputString + "\"" + " is not a valid unit for " + want)}
```

Comment:
I'm not particularly happy with the errors are either.
there is a third mode. 
-  The input was not a number at all. 

On the expanded error form I'm not sure how far we could go as far as correct usage of a flag.
I just put together a flagset to see the formatting. This is the output:
`
invalid value "10" for flag -freq: parse error: no units provided, need: "Hz"
`
I think it should look more like:
- The unit was missing:
`invalid value "10" for flag -freq: does not contain any units. Valid unit are Hertz or Hz`
- The unit was wrong:
`invalid value "10rpm" for flag -freq: incorrect usage. Valid unit are Hertz or Hz`
- The SI prefix was wrong:
`invalid value "10fHz" for flag -freq: could not correctly determine SI prefix. Valid prefixes are n, p, u, µ, m, k, M, G or T`
- Was not a number:
`invalid value "ten" for flag -freq: does not appear to be a number`

How does that look?

Reply:
I like the error messages.
My challenge: how do you distinguish between "unit was wrong" and "SI prefix was wrong" ?

Reply:
If there is a number a valid unit but with an extra Alpha runes that is not a prefix... 
My only thought atm on that is if gets to the default case and the prefix size is unit, there could be a check if any substrings are valid units. At one stage I was using and an acceptUnit(...string) bool that did that. Since it is only for a failing case, the extra cycles spent on the string comparisons would not matter? I'll code something up and see how it looks.

also some other errors need some tweaking for the user side.
`invalid value "+-10" for flag -freq: can't contain both plus and minus symbols`
`invalid value "--10" for flag -freq: can't contain multiple minus symbols`
`invalid value "++10" for flag -freq: can't contain multiple plus symbols`
`invalid value ".1.0" for flag -freq: can't contain multiple decimal points`
`invalid value "1□" for flag -freq: unexpected end of string`
`invalid value "-10THz" for flag -freq: exceeds minimum for unit of -9.23THz`
`invalid value "10THz" for flag -freq: exceeds maximum for unit of 9.23THz`

Reply:
Having slow(er) failing case is fine.